### PR TITLE
fix: rendering profile images (#480)

### LINKED ISSUE #480: Issue with Rendering the Profile Images.
### Description

Apparently, the image loading is quite not normal. Sometimes it gets loaded fully, sometimes it displays the alt text. Normally this should not happen, on a fully functional internet connection. 

### Screenshots

Screenshot 2021-10-08 at 22 36 01
Screenshot 2021-10-08 at 22 37 02

### Additional information

N/A

### PR DESCRIPTION
**Pull request by shashank-indukuri**
Resolve #480 

Changes:

1. Maintain global state for the profiles using Context API
2. Using useHistory hook, persist the state during the route from Socials to Home page.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/504"></a>

### CODE REVIEW

**Review by krishguptadev — CHANGES_REQUESTED** [review]
Please have a look at Eddie's comment 🙂

**Comment by emmalearnscode** [comment]
How’s it going with this issue @shashank-indukuri ? Please let us know if you need some guidance or help fixing merge conflicts

**Comment by shashank-indukuri** [comment]
I am so sorry. I am busy with my academics since last week. So, I haven't got enough time to look into it. I will work on it and will try to merge within 2 days. Thank you.

**Comment by emmalearnscode** [comment]
> I am so sorry. I am busy with my academics since last week. So, I haven't got enough time to look into it. I will work on it and will try to merge within 2 days. Thank you.

No worries! I just wanted to check how you were getting on. Studies come first of course.

**Review by krishguptadev — COMMENTED** [review]
## Merge Conflict

Thank you for your contribution. Your pull request currently has conflicting changes with our `main` branch. In order to review this pull request, we will need those conflicts to be resolved.

We recommend doing this with the `git rebase` command. If you are unsure of how to fix this, let us know and we would be happy to assist you.

**Review by krishguptadev — COMMENTED** [review]
## Checks Not Passing

Thank you for your pull request. Your contributions are unfortunately not passing our integration checks. Because of this, we are unable to review your pull request.

The errors can be addressed and resolved by:

If you have any questions about resolving this, feel free to ask us and we will gladly work with you.

**Comment by shashank-indukuri** [comment]
Thank you so much for the support. I will work on it and let you know if I need any help.

**Review by CodingSpecies — COMMENTED** [review]
Is there any updates @shashank-indukuri ? If you need help feel free to ask.
Thank you.

**Comment by krishguptadev** [comment]
Hey there, do you need any help?

**Comment by krishguptadev** [comment]
Yeh sure, feel free to drop the error here

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please see the comments above.

File: src/Components/Socials.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      })
+      .finally(() => setShowProgress(false))
+  }, [username])
+
+  return (
+    <main>
+      {showProgress && <ProgressBar mode="indeterminate" />}
+      {!showProgress && (
+        <>
+          {history.length === 2 ? (
```
Comment (Panquesito7) [inline src/Components/Socials.js:32]: I think there's an error here. 😅

image

Comment (shashank-indukuri) [reply src/Components/Socials.js]: I fixed it. Thanks.

**Comment by krishguptadev** [comment]
Hey there @shashank-indukuri 👋 Just checking in, do you need any help?

**Comment by shashank-indukuri** [comment]
> Hey there @shashank-indukuri 👋 Just checking in, do you need any help?

Hi @krishguptadev , Thanks for checking in. I have fixed it. Can you review the changes?

**Comment by eddiejaoude** [comment]
Thanks @shashank-indukuri for updating. Could you resolve the conflict too, thanks 👍

**Comment by shashank-indukuri** [comment]
I have resolved the conflicts. Thanks.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please revert the changes in the `package-lock.json` file.

**Review by emmalearnscode — COMMENTED** [review]
Why is context not being used in the socials.js file? If users have already been fetched and stored in context by app, shouldn't it be possible to filter that list by username instead of sending another fetch request? So in the useEffect an if check can look at if the list in context is already populated then another fetch doesn't need to be sent. If someone navigated directly to the page so there is no list then the fetch could be sent. Or have I misunderstood something? (My knowledge of context is not great)

**Comment by shashank-indukuri** [comment]
> Why is context not being used in the socials.js file? If users have already been fetched and stored in context by app, shouldn't it be possible to filter that list by username instead of sending another fetch request? So in the useEffect an if check can look at if the list in context is already populated then another fetch doesn't need to be sent. If someone navigated directly to the page so there is no list then the fetch could be sent. Or have I misunderstood something? (My knowledge of context is not great)

The list.json consists of all the users with only username, name and avatar fetched and stored in the context. So, the remaining details are not stored and need to get the profile-specific information every time one navigates to the specific profile.

**Comment by KendallDoesCoding** [comment]
> I have resolved the conflicts. Thanks.

There are still conflicts.. Please resolve
image

**Comment by emmalearnscode** [comment]
We have decided to go in another direction with regards to rendering the profile images on the homepage and therefore I will close this PR for now as there are still some conflicts that haven't been fixed. Thanks for your contribution and you are welcome  to come back and contribute in the future on other issues if you wish.

**Pull request closed without merging**